src/components/ActivityDetail/Reservation/Schedule.tsx

```diff
@@ -47,57 +46,50 @@ export default function Schedule(props: ScheduleProps) {
 
   const { bannerImageUrl, title, address, rating, price } = detailData;
 
-  const { data: userInfo } = useQuery({
-    queryKey: ['userInfo'],
-    queryFn: getUser,
-    staleTime: 0,
-  });
-
   const router = useRouter();
   const { activityid } = router.query;
   const activityId = Number(activityid);
 
-  const [loggedIn, setLoggedIn] = useState(false);
-  const [isModalOpen, setIsModalOpen] = useState(false);
   const [selectedScheduleId, setSelectedScheduleId] = useState<number | null>(
     null,
   );
 
+  const { data: userData } = useQuery({
+    queryKey: ['userInfo'],
+    queryFn: getUser,
+    staleTime: 0,
+    enabled: !!getCookie('refreshToken'),
+  });
+
   const bookableIds = new Set<number>();
   bookableSchedule.forEach((schedule) => {
     schedule.times.forEach((time) => {
       bookableIds.add(time.id);
     });
   });
 
-  const checkLoginState = () => {
-    const accessToken = getCookie('accessToken');
-    setLoggedIn(!!accessToken);
-  };
-
   const handleReservationClick = () => {
-    if (!loggedIn) {
+    if (!getCookie('refreshToken')) {
       notify('warning', '로그인이 필요한 서비스입니다.', () => {
         router.push('/login');
       });

```

Comment:
![chrome-capture-2024-7-23 (1)](https://github.com/user-attachments/assets/0a8160e3-0ed1-4256-80eb-48e48a9d0452)

onClose에 router.push 말고 밖으로 분리하는 게 나을 것 같아요! 토스트 반응이 좀 느려서 모달의 예약하기 버튼을 누르면 사용자가 warning 토스트를 보게 돼서 안 좋은 것 같습니다...! 토스트 띄움과 동시에 바로 login 페이지로 이동하는게 나을 것 같아유...!


```suggestion
if (~~~ ){
    notify('warning', '로그인이 필요한 서비스입니다.')
    router.push('/login');
}    
```